libcontainer/cgroups/systemd/common.go

```diff
@@ -194,16 +197,17 @@ func stopUnit(cm *dbusConnManager, unitName string) error {
 			return errors.New("Timed out while waiting for systemd to remove " + unitName)
 		}
 	}
+
+	// In case of a failed unit, let systemd remove it.
```

Comment:
I think most of time the unit could be removed succussfully by systemd, we only need this reset operation when failed, so maybe we should add a failed condition check here? Otherwise there is an unnecessary systemed operation than before in most of time.

Reply:
@lifubang thanks! I believe that checking the unit status will take the same amount of resources as doing reset-failed, maybe even more. This is why I am doing it unconditionally here.